app/contexts/Auth.tsx

```diff
@@ -0,0 +1,29 @@
+"use client";
+import { Auth0Provider } from "@auth0/auth0-react";
```

Comment:
React context is a client-only feature, which is where the change in mental model really hit home for me.

if want to use a 3rd-party context, you need to wrap it in "use client", unless the author does it for you. [Here are the nextjs docs](https://nextjs.org/docs/14/app/building-your-application/rendering/composition-patterns#using-third-party-packages-and-providers) that discuss that.